### PR TITLE
KAA-290: Added page with information about device provisioning and endpoint registration.

### PR DESCRIPTION
**Pull request by v-chyzhevskyi**
Added page with information about device provisioning and endpoint registration.

### CODE REVIEW

File: doc/Programming-guide/Key-platform-features/Devices-provisioning-and-registration/index.md [hunk 1]
```diff
@@ -6,5 +6,162 @@ nav: /:path/Programming-guide/Key-platform-features/Devices-provisioning-and-reg
...
+Credentials service may be configured separately for each application use either [Admin UI]() or [REST API](). 
+
+At the moment of writing Kaa endpoint credentials is identifier is based on hash of endpoint RSA Key Pair.
+During the start of new session, operations service lookup credentials service based on application identifier and uses this credentials service to lookup status of provided credentials object. 
+If Credentials object is not found or this credentials are in use by other endpoint, new session is rejected by Kaa node.
+
+### Endpoint Registration information
+
+Kaa platform user is able to provision server-side profile for new endpoints using information about endpoint credentials (See following [REST API](TODO) call for more details). 
+This server-side profile will be used during endpoint registration.
```
Comment (balldir) [inline doc/Programming-guide/Key-platform-features/Devices-provisioning-and-registration/index.md:65]: Ticket number for TODO ?

File: doc/Programming-guide/Key-platform-features/Devices-provisioning-and-registration/index.md [hunk 2]
```diff
@@ -6,5 +6,162 @@ nav: /:path/Programming-guide/Key-platform-features/Devices-provisioning-and-reg
...
+    <entry key="NEW" value-ref="customCredentialsServiceLocator"/>                          
+</util:map>
+
+```
+
+Value of key which was added you will see in Admin UI.
+
+![credential](attach/credential.png)
+
+3. In in /usr/lib/kaa-node/conf/common-dao-context.xml
```
Comment (balldir) [inline doc/Programming-guide/Key-platform-features/Devices-provisioning-and-registration/index.md]: > In in 

???

File: doc/Programming-guide/Key-platform-features/Devices-provisioning-and-registration/index.md (hunk 1 above)
Comment (v-chyzhevskyi) [inline doc/Programming-guide/Key-platform-features/Devices-provisioning-and-registration/index.md:65]: http://jira.kaaproject.org/browse/KAA-290

**Comment by ashvayka** [comment]
LGTM

**Comment by balldir** [comment]
LGTM

**Pull request merged**